Add CITATION.cff and DOI badge

Ly0n opened the pull request:
## Changes proposed in this Pull Request

Hi PyPSA-Earth folks :wave:,
I noticed that there is no link to the peer-reviewed scientific paper on PyPSA-Earth. It is also not clear how to cite this repository. Therefore, I would recommend mentioning the first publication and the corresponding DOI in the README and in a CITATION.CFF. This can help to better highlight the origin, quality and impact of the project. The CITATION.CFF file has been automatically generated from this website: https://cff-editor.mcrespo.me/ 
ORCID links have been added manually by me for the people I could find an account for.

ekatef commented:
Hey @Ly0n, a definitely great idea! Thanks a lot for that 😄

I see that REUSE linter is complaining. Would you mind to add a license header to `CITATION.cff`?

Ly0n commented:
Hey @ekatef,

I fixed the badge and added the header. The "Cite this repository" is now visible on the right side and the badge also looks good now. See here-> https://github.com/Ly0n/pypsa-earth/tree/add-CITATION-cff

Ly0n commented:
@ekatef Done! :)

ekatef commented:
> @ekatef Done! :)

Perfect, thank you! Merged 😄 
Great to have you in the contributors list 🙂
